### ISSUE TITLE
Accessing std::env::args from a Rust library in a C++ binary fails on FreeBSD

### ISSUE THREAD

**Issue by zanchey** [dtolnay/cxx#1306]
I'm not sure that this is definitely a cxx problem, but here's the investigation I've done so far.

`std::env::args` when called from a Rust library, linked to a C++ binary, returns an empty array on FreeBSD.

There's a minimal reproducing repo at https://github.com/zanchey/argstest/ (clone and run make to produce an argstest binary) which:

- has a C++ entrypoint, that
- calls a Rust library function, that
- prints the contents of `std::env::args`

On Linux/glibc this correctly produces (say) `Args { inner: ["./argstest", "foo", "bar", "baz"] }`, but on FreeBSD returns `Args { inner: [] }`.

I don't know if cxx does any magic with the args vector to make it visible from Rust? I couldn't see anything obvious.

**Comment by dtolnay** [comment]
I don't think this has to do with anything in this crate. I expect that you would observe the same behavior for any other way that a C++ binary would end up executing Rust library code without a Rust `main`, including handwritten C FFI.
